### ISSUE TITLE
camel-undertow consumer endpoints with transform expressions fail on redployment

### ISSUE THREAD

**Issue by jamesnetherton** [wildfly-extras/wildfly-camel#987]
This route causes some problems when using CDI:

```java
@ContextName("cdi-context")
public class MyRouteBuilder extends RouteBuilder {
    @Override
    public void configure() throws Exception {
        from("undertow://http://localhost/test/hello").transform(simple("Hello ${header.name}!"));
    }
}
```
When the application is first deployed it works fine. Redeploy it onto a running application server and the HTTP endpoint will fail with the following exception. There is some strange issue with resolving the camel simple language beans.
```
org.jboss.weld.exceptions.IllegalStateException: WELD-001333: BeanManager method getBeans() is not available after shutdown
	at org.jboss.weld.bean.builtin.BeanManagerProxy.checkContainerState(BeanManagerProxy.java:239)
	at org.jboss.weld.bean.builtin.BeanManagerProxy.checkContainerState(BeanManagerProxy.java:247)
	at org.jboss.weld.bean.builtin.BeanManagerProxy.getBeans(BeanManagerProxy.java:96)
	at org.apache.deltaspike.core.api.provider.BeanProvider.getContextualReference(BeanProvider.java:144)
	at org.apache.deltaspike.core.api.provider.BeanProvider.getContextualReference(BeanProvider.java:121)
	at org.apache.camel.cdi.CdiInjector.newInstance(CdiInjector.java:44)
	at org.apache.camel.impl.DefaultLanguageResolver.resolveLanguage(DefaultLanguageResolver.java:74)
	at org.apache.camel.impl.DefaultCamelContext.resolveLanguage(DefaultCamelContext.java:2203)
	at org.apache.camel.builder.SimpleBuilder.createExpression(SimpleBuilder.java:101)
	at org.apache.camel.builder.SimpleBuilder.evaluate(SimpleBuilder.java:82)
	at org.apache.camel.processor.TransformProcessor.process(TransformProcessor.java:50)
	at org.apache.camel.management.InstrumentationProcessor.process(InstrumentationProcessor.java:77)
	at org.apache.camel.processor.RedeliveryErrorHandler.process(RedeliveryErrorHandler.java:460)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:190)
	at org.apache.camel.processor.CamelInternalProcessor.process(CamelInternalProcessor.java:190)
	at org.apache.camel.util.AsyncProcessorHelper.process(AsyncProcessorHelper.java:109)
	at org.apache.camel.processor.DelegateAsyncProcessor.process(DelegateAsyncProcessor.java:87)
	at org.apache.camel.component.undertow.handlers.HttpCamelHandler.handleRequest(HttpCamelHandler.java:97)
	at io.undertow.server.Connectors.executeRootHandler(Connectors.java:199)
	at io.undertow.server.HttpServerExchange$1.run(HttpServerExchange.java:774)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

**Comment by jamesnetherton** [comment]
This problem is probably not specific to the simple language but also other expression languages. Although `transform(constant("hello world"))` works fine.

**Comment by jamesnetherton** [comment]
This looks like a possible issue with DeltaSpike. If I use the [new and improved camel-cdi component](https://github.com/astefanutti/camel-cdi/) (which does not depend on DeltaSpike), this problem is not present.

Camel 2.17 will hopefully ship with the new camel-cdi, so I'll move this issue to the 4.0 milestone. The work to contribute the new camel-cdi component to camel is tracked here https://github.com/astefanutti/camel-cdi/issues/7.

**Comment by jamesnetherton** [comment]
Closing this. It's resolved in Camel 2.17 and there's no work required on the subsystem side.
